### PR TITLE
ISSUE-31 update on window resize and path change

### PR DESCRIPTION
**Pull request by simonhearne**
I ended up implementing an interval as one was required to check for path changes anyway, it also handily debounces the resize.
onHashChange doesn't always work e.g. <https://www.oliverbonas.com/> is a SPA that updates the path, not hash. Set it at 500ms which seems like a reasonable compromise.

### CODE REVIEW

File: app/public.js [hunk 1]
```diff
@@ -70,13 +72,50 @@
...
     });
+    setEventListeners();
   }
 
   /**
+   * Refresh overlay
+   */
+  function refreshImagesInfo() {
+    hideImagesInfo();
+    showImagesInfo(null);
```
Comment (tabanliviu) [inline app/public.js]: @simonhearne DoD - why send null? there is no need. undefined is good enough

File: app/public.js [hunk 2]
```diff
@@ -70,13 +72,50 @@
...
+  /**
+   * Set event listeners
+   */
+  function setEventListeners() {
+    let width = window.innerWidth;
+    let height = window.innerHeight;
+    let path = window.location.pathname;
+    let update;
+
+    interval = setInterval(function() {
```
Comment (tabanliviu) [inline app/public.js]: @simonhearne CONSISTENCY - `setInterval(() => {})`

Comment (tabanliviu) [inline app/public.js]: @simonhearne DoD - should clear interval before creating a new one as we're always create another interval every time showImagesInfo is called

File: app/public.js [hunk 3]
```diff
@@ -21,6 +21,8 @@
   const SVG_URL_REGEX = /\.svg/i;
   const SVG_DATA_URI_REGEX = /^data:.*\/svg/i;
   const MIN_IMAGE_SIZE = 10;
+  const CHECKER_INTERVAL_MS = 500;
+  let interval;
```
Comment (tabanliviu) [inline app/public.js]: @simonhearne MAINTENANCE - use a more descriptive name like `refreshImagesInterval`

**Review by tabanliviu — CHANGES_REQUESTED** [review]

File: app/public.js (hunk 3 above)
Comment (simonhearne) [reply app/public.js]: Yup, done.

File: app/public.js (hunk 1 above)
Comment (simonhearne) [reply app/public.js]: Fixed.

File: app/public.js (hunk 2 above)
Comment (simonhearne) [reply app/public.js]: It's cleared in hideImagesInfo(). Assume we never call showImagesInfo() twice without hideImagesInfo()? Then we would get duplicate overlays. Could add a check in showImagesInfo() for isImagesInforActive() then hide first... new issue for that?

**Review by simonhearne — COMMENTED** [review]
Updated branch to include the requested changes.

File: app/public.js (hunk 2 above)
Comment (tabanliviu) [reply app/public.js]: the idea is to not depend on a different method for clearing the interval. this couples those 2 methods together. in the case of calling showImagesInfo twice, it should be safe, including clearing existing overlays and existing intervals.

perhaps showImagesInfo should always call hideImagesInfo at the beginning to ensure a clean slate.

**Comment by tabanliviu** [comment]
@simonhearne still need to add unit tests :smile:

**Comment by tabanliviu** [comment]
fixes #31

**Review by tabanliviu — APPROVED** [review]

**Comment by Ian-Hi** [comment]
Verified ok

File: app/public.js [hunk 4]
```diff
@@ -70,13 +72,50 @@
...
+    let height = window.innerHeight;
+    let path = window.location.pathname;
+    let update;
+
+    refreshImagesInterval = setInterval(() => {
+      update = false;
+      /* istanbul ignore if */
+      if (window.location.pathname !== path) {
+        update = true;
+        path = window.location.pathname;
```
Comment (MalcolmSmithUK) [inline app/public.js]: @tabanliviu DoD - This doesn't include changes in the hash (either for traditional hash or non-HTML5 SPAs).

File: app/public.js [hunk 5]
```diff
@@ -70,13 +72,50 @@
...
+   */
+  function refreshImagesInfo() {
+    hideImagesInfo();
+    showImagesInfo();
+  }
+
+  /**
+   * Set event listeners
+   */
+  function setRefreshImagesWatcher() {
```
Comment (MalcolmSmithUK) [inline app/public.js:89]: @tabanliviu MAINTAINABILITY - Please clear the timeout in case another event implements the same method.

**Review by MalcolmSmithUK — CHANGES_REQUESTED** [review]

File: app/public.js (hunk 5 above)
Comment (tabanliviu) [reply app/public.js:89]: OK

File: app/public.js (hunk 4 above)
Comment (tabanliviu) [reply app/public.js]: OK

**Comment by tabanliviu** [comment]
@MalcolmSmithUK ready for review

**Review by MalcolmSmithUK — APPROVED** [review]

**Pull request merged**